src/caustics/lenses/pixelated_convergence.py

```diff
@@ -145,7 +140,10 @@ def __init__(
         self.add_param("y0", y0)
         self.add_param("convergence_map", convergence_map, shape)
 
-        self.n_pix = n_pix
+        if convergence_map is not None:
```

Comment:
Should probably throw an error if convergence_map and shape are both missing

Reply:
Hmm, this error is caught above in line 132-138 roughly.